### ISSUE TITLE
Inquiry about the detailed config of reproduction

### ISSUE THREAD

**Issue by fzhiy** [NVlabs/QeRL#14]
Hi, authors,

Thank you very much for your excellent work. I believe this will undoubtedly advance further research in practical reinforcement learning.

I am attempting to reproduce the results of QeRL-supported GRPO on Qwen 2.5-3B using an RTX 5090. I trained the base model quantized via NVFP4 using the hyperparameters from Table 4 in the paper (max_completion_length=4096, max_seq_length=5500, num_iterations=4, lora_rank=32). When evaluated on GSM8K, the accuracy was only 66.5%, which is significantly lower than the 83.7% reported in Table 1(a).

What could be causing this discrepancy? Have I overlooked any critical parameter configurations?
If possible, could you share the detailed configuration file used to reproduce the main results in Table 1 and Table 2?

Thank you in advance.

Best regards,
Feng

**Comment by oncleJules** [comment]
I have the same concern.

The scripts' filenames started with 'dapo', does it mean the hyperparameters in these scripts are only for dapo, not for grpo?

since the experiments of grpo are significantly shorter, i believe most practitioners will try grpo experiments first - would greatly appreciate a release of the grpo hyperparameters

**Comment by Aaronhuang-778** [comment]
@oncleJules 

Thanks for your remind. We will release both optimal  setting of GRPO and DAPO scripts.

**Comment by Aaronhuang-778** [comment]
> Hi, authors,
> 
> Thank you very much for your excellent work. I believe this will undoubtedly advance further research in practical reinforcement learning.
> 
> I am attempting to reproduce the results of QeRL-supported GRPO on Qwen 2.5-3B using an RTX 5090. I trained the base model quantized via NVFP4 using the hyperparameters from Table 4 in the paper (max_completion_length=4096, max_seq_length=5500, num_iterations=4, lora_rank=32). When evaluated on GSM8K, the accuracy was only 66.5%, which is significantly lower than the 83.7% reported in Table 1(a).
> 
> What could be causing this discrepancy? Have I overlooked any critical parameter configurations? If possible, could you share the detailed configuration file used to reproduce the main results in Table 1 and Table 2?
> 
> Thank you in advance.
> 
> Best regards, Feng

Hi, please just set larger batch size for the training. 

The scripts provided in `training` path are for just for limited resource training. In our experiments, we used larger batch-size as we can in H100, which provided stable results. Hope this will be helpful :-) .

**Comment by oncleJules** [comment]
> > I ran into this problem as well. In my replication, the Qwen 2.5 7B NVFP4 baseline score was 0.71, and the score only improved to 0.73 after complete QeRL training. I used the project's evaluation script.
> > Another question, which prompt was used for the GSM8K training and eval in the paper? I see there are three prompts here, with two commented out. I've noticed that the structure and placement order of the system prompt significantly affects the score.
> > https://github.com/NVlabs/QeRL/blob/main/utils/data.py#L16-L49
> > ```
> > # uncomment middle messages for 1-shot prompting
> > def get_gsm8k_questions(split = "train") -> Dataset:
> >     data = load_dataset('openai/gsm8k', 'main')[split] # type: ignore
> >     # system_prompt = """
> >     #     Solve the following math problem step by step. The reasoning process and direct answer are enclosed within <think> </think> and <answer> </answer> tags, respectively, i.e., <think> reasoning process here </think> <answer> answer here </answer>: 
> >     #     <think>
> >     #     ...
> >     #     </think>
> >     #     <answer>
> >     #     ...
> >     #     </answer>
> >     # """
> >     # system_prompt = """
> >     #     Solve the following math problem step by step. The reasoning process is enclosed within <think> </think> tags, and the final answer is provided after "####", i.e.,: 
> >     #     <think>
> >     #     ...
> >     #     </think>
> >     #     #### answer
> >     # """
> >     instruction = """
> >         Let\'s think step by step first within <think> </think> tags, and output the final answer after "####" tag, i.e.,: 
> >         #     <think>
> >         #     ...
> >         #     </think>
> >         #     #### number
> >     """
> >     data = data.map(lambda x: { # type: ignore
> >         'prompt': [
> >             # {'role': 'system', 'content': system_prompt},
> >             {'role': 'user', 'content': x['question']  + " " + instruction}
> >         ],
> >         'answer': extract_hash_answer(x['answer'])
> >     }) # type: ignore
> >     return data # type: ignore
> > ```
> 
> Hi, thank you for your attention of our work. We use the first system prompt for GSM8K:
> 
> <img alt="Image" width="2000" height="334" src="https://private-user-images.githubusercontent.com/71201052/516263333-6b69f878-9c32-412c-b920-f7fe71b38784.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjU1OTIwMTYsIm5iZiI6MTc2NTU5MTcxNiwicGF0aCI6Ii83MTIwMTA1Mi81MTYyNjMzMzMtNmI2OWY4NzgtOWMzMi00MTJjLWI5MjAtZjdmZTcxYjM4Nzg0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMjEzVDAyMDgzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU1MTgwY2I1ZjgyMmZhMTQ2YmFiMzU1Y2YwNGZmZjQ2ZWJkYWNkOTFkOGRhMjY4MDc0OGJlOWRhZWE2YmNmZDImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ZiLO5uPL6jXu8KF-9dH3PzqBaZCFtDOiu1gOlMa6LpY">
> The script we provided is just used for resource-limited single GPU, it is for training reward/ efficiency comparison with BF16 + LoRA, you can also train BF16 + LoRA for comparing. In our manuscript's table, we used H100-80G and set larger batch-size (batch_size 64*8). It is because that larger batch-size brings stable and batter training results. You can change `perdevice_train_batch_size` and `gradient_accumulation_steps` on single GPU to get stable training results, and we will also release the multi-GPU training version and scripts later (currently struggling for rebuttal :-) )

Hi @Aaronhuang-778 , hopefully the rebuttal has been completed successfully. Would the configs be released soon?
